Added Casting Channels

KnightMiner opened the pull request:
![2017-10-26_18 58 37](https://user-images.githubusercontent.com/12155288/32082513-a98c61fc-ba81-11e7-953e-62309cb3114a.png)

Adds back casting channels. Channels no longer dynamically choose flow direction, instead setting the direct on placement and through right click.

Some additional information on mechanics:

* Channels will automatically connect to the block they are placed on, but not any other sides
* Channels cannot extract liquids. Instead, a channel is typically filled using a faucet.
* Channels will prioritize downwards flowing if possible. If they cannot flow down, they will split evenly between all sides.
* Channels are redstone sensitive. Toggling redstone will toggle whether the downwards connection is enabled

For a bit more information, I have a writeup on the [Ceramics Wiki](https://github.com/KnightMiner/Ceramics/wiki/Channel)

Landmaster commented on src/main/java/slimeknights/tconstruct/smeltery/network/ChannelConnectionPacket.java:
```diff
@@ -0,0 +1,53 @@
+  @Override
+  public void handleServerSafe(NetHandlerPlayServer netHandler) {
+    // clientside only
+    throw new UnsupportedOperationException("Serverside only");
```
Should not this be `"Clientside only"`?

KnightMiner replied:
That's based on the fluid update packet. I assume boni intended it to mean serverside only to send it, but it could have been a typo there

bonii-xx commented:
Errmagerrrrrrrd

bonii-xx commented:
Ok, so from a function wise perspective they work pretty well. A few things I noticed:

- I built a channel, and then wanted to put a tank below it. The block below the channel was free. It seems to have tried to toggle the downflow, so i couldn't place the tank there - Behaves a bit weird there.
- The indications for the flow direction aren't clear. I can't tell which way is which. Instead of a 4x4 area in the middle of the connection I'd rather go for a more arrow shaped mark (see below)
- It'd be nice to also have some kind of indication on the outside, or maybe when you hover over the block
- WAILA/TOP don't seem to display any content information. If the fluid is stuck you don't know which fluid and how much is in there.


Super fancy ASCII art for the marker:
Current:
OOOOO
OXXXXO
OXXXXO
OOOOO

Suggested for flowing "left":
OOXXO
OXXOO
OXXOO
OOXXO

KnightMiner commented:
Fixed the first issue and the last issue for TOP (WAILA is being stubborn, nut sure why its not working there)

Also added a new texture for the top of channels, as shown below:
![2017-11-26_12 47 54](https://user-images.githubusercontent.com/12155288/33243307-2831751c-d2a9-11e7-8a01-6ce03b9328f0.png)
Not sure what to do for an indicator on the sides since I have so little texture space to work with.

laz2727 commented:
OQOO
QQQQ

On Nov 26, 2017 23:56, "KnightMiner" <notifications@github.com> wrote:

> Fixed the first issue and the last issue for TOP (WAILA is being stubborn,
> nut sure why its not working there)
>
> Also added a new texture for the top of channels, as shown below:
> [image: 2017-11-26_12 47 54]
> <https://user-images.githubusercontent.com/12155288/33243307-2831751c-d2a9-11e7-8a01-6ce03b9328f0.png>
> Not sure what to do for an indicator on the sides since I have so little
> texture space to work with.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SlimeKnights/TinkersConstruct/pull/3213#issuecomment-347029978>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACQo19D3zh5Dh4auwsQLyi0P4cCFeW3gks5s6bRsgaJpZM4QIYxT>
> .
>

KnightMiner commented:
Hard part is thats a bit vague as now left vs right is a 2 pixel difference. Then again, its better than nothing.
